File: DummyAPI/Startup.cs
using DummyAPI.Middlewares;
using DummyAPI.Persistence;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;
using Prometheus;
using Prometheus.Advanced;
using Swashbuckle.AspNetCore.Swagger;
using System;

namespace DummyAPI
{
    public class Startup
    {
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddMvc().AddJsonOptions(options => {
                options.SerializerSettings.ContractResolver = new CamelCasePropertyNamesContractResolver();
                options.SerializerSettings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore;
            });

            services.AddDbContext<BloggingContext>(options => options.UseSqlite("Data source=..\\dummy.db"));

            services.AddSwaggerGen(c =>
            {
                c.SwaggerDoc("dummy", new Info { Title = "Dummy API" });
            });

            services.AddSingleton<ICollectorRegistry>(DefaultCollectorRegistry.Instance);
        }

        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }

            app.Use((ctx, x) =>
            {
                Console.WriteLine(ctx.Request.Path);
                return x.Invoke();
            });

            app.UseMetricServer();
            app.UseMiddleware<ResponseTimeMiddleware>();
            app.UseMiddleware<StatusCodeMiddleware>();

            app.UseCors(builder => {
                builder.AllowAnyOrigin();
            });

            app.UseSwagger();
            app.UseSwaggerUI(c =>
            {
                c.SwaggerEndpoint("/swagger/dummy/swagger.json", "Dummy API");
            });


            app.UseMvc();
        }
    }
}

File: DummyAPI.FSharp/Startup.fs
namespace DummyAPI.FSharp

open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Hosting
open Microsoft.AspNetCore.Http
open Microsoft.Extensions.DependencyInjection
open Newtonsoft.Json.Serialization
open Newtonsoft.Json
open Microsoft.EntityFrameworkCore
open Persistence
open Swashbuckle.AspNetCore.Swagger

type Startup() =

    member this.ConfigureServices(services: IServiceCollection) =

        services
            .AddMvc()
            .AddJsonOptions(fun options ->
                options.SerializerSettings.ContractResolver <- CamelCasePropertyNamesContractResolver()
                options.SerializerSettings.ReferenceLoopHandling <- ReferenceLoopHandling.Ignore
            ) |> ignore

        services.AddDbContext<BloggingContext>(fun options ->
            options.UseSqlite("Data source=..\\dummy.db") |> ignore
        ) |> ignore


        services.AddSwaggerGen(fun c ->
            c.SwaggerDoc("dummy", Info(Title = "Dummy API")) |> ignore
        ) |> ignore

        ()

    member this.Configure(app: IApplicationBuilder, env: IHostingEnvironment) =
        if env.IsDevelopment() then
            app.UseDeveloperExceptionPage() |> ignore

        app.UseSwagger()
        |> ignore

        app.UseSwaggerUI(fun c ->
            c.SwaggerEndpoint("/swagger/dummy/swagger.json", "Dummy API")
        ) |> ignore


        app.UseMvc();